synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -318,6 +323,12 @@ def _download_url(self, url, user):
                 length, headers, uri, code = yield self.client.get_file(
                     url, output_stream=f, max_size=self.max_spider_size,
                 )
+            except SynapseError:
+                # Pass SynapseErrors through directly, so that the servlet
+                # handler will return a SynapseError to the client instead of
+                # blank data or a 500. Currently, this is only if the IP we are
```

Comment:
> Currently, this is only if the IP we are
> trying to fetch from is blacklisted.

This really isn't true - `get_file` can raise a SynapseError for plenty of other reasons.

That's not to say that what you're doing here is wrong, but I find the last sentence of the comment misleading. Could you get rid of it?

Reply:
yeah actually that's fair, it does it on sizes and things too.